src/main/kotlin/se/uulm/snowballr/backend/service/UserService.kt

```diff
@@ -187,4 +193,32 @@ class UserService(
         val updatedUser = userRepo.updateUser(request)
         return updatedUser.toGrpcUser()
     }
+
+    override suspend fun softDeleteUser(request: Base.Id): Base.Nothing {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+        val userToDelete = userRepo.getUserById(parseUUID(request.id, EntityType.USER))
+        val isSameUser = currentUser.id == userToDelete.id
+
+        // Checks, if the user tries to delete another user without being an admin
+        if (currentUser.role != UserRole.USER_ROLE_ADMIN && !isSameUser) {
+            throw UnauthorizedException.Single(
+                EntityType.USER,
+                userToDelete.id.toString(),
+                AccessType.DELETE,
+                currentUser.id.toString(),
+                IdentifierType.ID,
+            )
+        }
```

Comment:
This helper method has been added since the last review.
```suggestion
        if (!isSameUser) {
            verifyServerAdminRole(currentUser) {
                UnauthorizedException.Single(EntityType.USER, userToDelete.id.toString(), AccessType.DELETE, it)
            }
        }

```

Reply:
Good catch.